src/app/shared/card/card.component.html

```diff
@@ -1 +1,9 @@
+<a routerLink="/detail/{{hero.id}}" >
```

Comment:
use child parent communicator to decide what to do on click of the card, this will be too specific